packages/beeq/src/components/icon/helper/request.ts

```diff
@@ -2,38 +2,41 @@
 /*                             Icon request helper                            */
 /* -------------------------------------------------------------------------- */
 
-import { isString } from '../../../shared/utils';
+import { isNil, isString } from '../../../shared/utils';
 
 const requests = new Map<string, Promise<unknown>>();
 
 const fetchSvg = async (url: string, sanitize: boolean): Promise<unknown> => {
-  let req: Promise<unknown>;
-
-  if (typeof fetch !== 'undefined' && typeof document !== 'undefined') {
-    const rsp = await fetch(url);
-    if (rsp.ok) {
-      return rsp.text().then((svgContent) => {
-        if (svgContent && sanitize !== false) svgContent = validateContent(svgContent);
-        iconContent.set(url, svgContent || '');
-      });
-    }
+  if (typeof fetch === 'undefined' || typeof document === 'undefined') return;
+
+  if (requests.has(url)) return requests.get(url);
+
+  const response = await fetch(url);
+
+  if (!response.ok) {
     iconContent.set(url, '');
-    // cache for the same requests
-    requests.set(url, req);
-    return req;
+    return;
   }
 
-  iconContent.set(url, '');
-  return Promise.resolve();
+  const svgContent = await response.text();
+
+  if (svgContent && sanitize !== false) {
+    iconContent.set(url, validateContent(svgContent));
+  } else {
+    iconContent.set(url, svgContent || '');
+  }
```

Comment:
Perhaps we can improve this method by:
* reducing the complexity of the `if` by using a ternary operator
* a try/catch to handle potential errors during the fetch

Can you check if something like this could work?

```ts
const fetchSvg = async (url: string, sanitize: boolean): Promise<unknown> => {
  if (typeof fetch === 'undefined' || typeof document === 'undefined') return;

  if (requests.has(url)) return requests.get(url);

  try {
    const response = await fetch(url);

    if (!response.ok) {
      iconContent.set(url, '');
      return;
    }

    const svgContent = await response.text();
    const content = svgContent && sanitize !== false ? validateContent(svgContent) : svgContent || '';
    iconContent.set(url, content);
  } catch (error) {
    console.error(`Failed to fetch SVG from ${url}:`, error);
    iconContent.set(url, '');
  }
};
```

Reply:
If we want to reduce complexity I would go for 
```typescript
  try {
    const response = await fetch(url);

    if (!response.ok) {
      iconContent.set(url, '');
      return;
    }

    let svgContent = (await response.text()) || '';

    if (svgContent && sanitize !== false) svgContent = validateContent(svgContent);

    iconContent.set(url, svgContent);
  } catch (error) {
    console.error(`[BqIcon] Failed to fetch SVG from ${url}:`, error);
    iconContent.set(url, '');
  }
```